---
path: "/analyze/portals/analysis-portals/ucsc-xena"
date: "2019-02-05"
title: "UCSC Xena"
author: "Brian Craft, Mary Goldman, Jingchun Zhu, David Haussler"
description: "UCSC Xena is an online exploration tool for multi-omic functional genomics data and associated meta-data and annotations."
appUrl: "https://singlecell.xenahubs.net"
githubUrl: "https://singlecell.xenahubs.net"
componentName: "analysisDetail"
---

[UCSC Xena single cell browser](https://singlecell.xenahubs.net) is an online exploration tool for single cell RNA-seq data and associated meta-data and annotations.
 
 Xena is built for performance, allowing you to dynamically view 1 million cells at a time. Xena downloads the latest HCA gene expression matrices available at [HCA Data Portal](/) and makes them available for visualization via our Visual Spreadsheet. 

[![UCSC Xena single cell browser](../_images/portals/ucsc-xena.png)](https://singlecell.xenahubs.net)

Xena's Visual Spreadsheet is analogous to an office spreadsheet: it is a visual representation of a data grid where each column is a slice of genomic data (e.g. a gene or a set of genes' expression, inferred cell type, cell location), and each row is a single cell.
 
 Users interactively build their spreadsheet, enabling discovery across multiple genes and cell annotations. Researchers can dynamically combine their own data, such as cluster calls, with the data from the HCA while still keeping their data private.



## Contact
[ucsc-cancer-genomics-browser@googlegroups.com](mailto:ucsc-cancer-genomics-browser@googlegroups.com)\
[genome-cancer@soe.ucsc.edu](mailto:genome-cancer@soe.ucsc.edu)\
Follow [@UCSCXena](https://twitter.com/UCSCXena)

## Source code
<a href="https://github.com/ucscXena">https://github.com/ucscXena</a>
