---
path: "/contribute/data/contributing-to-hca/contributing-data-processing-results"
date: "2019-04-29"
title: "Data Processing and Results"
description: "Overview of processing of data submissions."
---

# Data Processing and Results

Once submitted to the Human Cell Atlas, your data will be processed and made available in multiple ways. 
 
## Searchable in the HCA Data Browser

Your raw data and metadata will be searchable in the <link-to-browser relativeLink="/projects">HCA Data Browser</link-to-browser> immediately after submission.

## Processed by the Data Processing Pipelines

Some submitted data will be processed by our standard [data processing pipelines](/pipelines). Currently, we have pipelines for processing human, single-cell and single-nuclei RNA-seq data produced by standard Smart-seq2 and 10X 3' v2 and v3 protocols. These pipelines will be automatically run immediately after submission, and the alignment, quantification, and quality control metric results will be made available in the HCA Data Browser alongside the raw data.


## Accessioned in Standard Archives

To support publishing your research, we will submit raw data to community standard archives and return the archival accessions that you need for publication.

Currently we are able to archive projects, samples, and sequencing data in BioStudies, BioSamples, and the European Nucleotide Archives (ENA), respectively. For managed-access data, we archive in the European Genome-phenome Archive (EGA). We are working towards supporting submission to the NCBI Gene Expression Omnibus (GEO) if your funding mandates submission there. Please let a data wrangler know if this is the case when you start your submission process.

At the moment we will not automatically archive imaging data. We are engaged with the community to understand what the archival best practices are for biological images, and we will work to support submission to these archives in the future.
