---
description: "Data contributors wishing to submit datasets to HCA for integration into atlases are required to submit an AnnData file."
title: "Submitting Count Matrices and Tier 1 Metadata to CELLxGENE Discover"
---

# Submitting Count Matrices and Tier 1 Metadata to CELLxGENE Discover

## File format

Data contributors wishing to submit datasets to HCA for integration into atlases are required to submit an AnnData file. [AnnData files](https://anndata.readthedocs.io/en/latest/index.html) combine matrices and metadata into a single file, thus metadata should be captured on a per-cell basis.

An AnnData file has several components, including:

- uns (Dataset metadata), which describe the dataset as a whole
- X (Matrix layers), which describe the data required for different assays
- obs (Cell metadata), which describe each cell in the dataset
- obsm (Embeddings), which describe each embedding in the dataset
- obsp, which describe pairwise annotation of observations
- var and raw.var (Gene metadata), which describe each gene in the dataset
- varm, which describe multidimensional annotation of variables/features
- varp, which describe pairwise annotation of variables/features

## File format specifications

- **Dataset-level metadata in uns:**
    - title
    - study_pi
    - batch_condition
    - default_embedding
    - comments


- **Data in .X and raw.X:**
    - raw counts are required
    - normalized counts are strongly recommended
    - raw counts should be in raw.X if normalized counts are in .X
    - if there is no normalized matrix, raw counts should be in .X


- **Cell metadata in obs:** Metadata fields are to be captured on a <ins> **per-cell** </ins> basis.

<StyledTable borderless noWrap>

|                               |                          |
|:--------------------------------------|:-----------------------------------|
| <li>protocol_url</li>                       | <li>manner_of_death</li>                    |
| <li>donor_id</li>                              | <li>sample_source</li>                      |
| <li>sample_id</li>                             | <li>sex_ontology_term_id</li>               |
| <li>institute</li>                             | <li>sample_collection_method</li>           |
| <li>sample_collection_site</li>                | <li>tissue_type</li>                        |
| <li>sample_collection_relative_time_point</li> | <li>sampled_site_condition</li>             |
| <li>library_id</li>                            | <li>tissue_ontology_term_id</li>            |
| <li>library_id_repository</li>                 | <li>tissue_free_text</li>                   |
| <li>author_batch_notes</li>                    | <li>sample_preservation_method</li>         |
| <li>organism_ontology_term_id</li>             | <li>suspension_type</li>                    |
| <li>cell_enrichment</li>                       | <li>cell_viability_percentage</li>          |
| <li>cell_number_loaded</li>                    | <li>sample_collection_year</li>             |
| <li>assay_ontology_term_id</li>                | <li>library_preparation_batch</li>          |
| <li>library_sequencing_run</li>                | <li>sequenced_fragment</li>                 |
| <li>sequencing_platform</li>                   | <li>is_primary_data</li>                    |
| <li>reference_genome</li>                      | <li>gene_annotation_version</li>            |
| <li>alignment_software</li>                    | <li>intron_inclusion</li>                   |
| <li>author_cell_type</li>                      | <li>cell_type_ontology_term</li>            |

</StyledTable>

In addition to the HCA obs fields above, there are an additional three fields that are required for submission into CELLxGENE. These fields are not part of the HCA Tier 1 metadata fields.
    - [disease_ontology_term_id](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md#disease_ontology_term_id)
    - [development_stage_ontology_term_id](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md#development_stage_ontology_term_id)
    - [self_reported_ethnicity_ontology_term_id](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md#self_reported_ethnicity_ontology_term_id)

- **Embeddings in obsm:**
    - One or more two-dimensional embeddings, prefixed with 'X_'

- **Features in var & raw.var (if present):**
    - index is Ensembl ID
    - preference is that gene have not been filtered in order to maximise future data integration efforts

<Alert icon={false} severity="info">
Should you require support creating an AnnData file or converting your file from another single cell file format click [here](https://github.com/cellgeni/sceasy) or reach out to [cellxgene@chanzuckerberg.com](mailto:cellxgene@chanzuckerberg.com) and mention that you are contributing to the HCA.
</Alert>

## Tier 1 metadata fields

Tier 1 metadata fields provide the foundational information used to build tissue and organ atlases. Specifically, the fields help to identify factors that can cause ‘batch effects’ that may arise when combining datasets into an atlas.

The full list of fields with detailed information on each field can be found [here](https://docs.google.com/spreadsheets/d/13oqRLh1awe7bClpX617_HQaoS8XPZV5JKPtPEff8-p4/edit?gid=1404414727#gid=1404414727). Some fields are are mandatory (labelled ‘must’) whereas others are ‘recommended’.

## Submission process

AnnData files will be stored and are accessible on CELLxGENE Discover. To submit files, please follow the process for submission:

1. Please reach out to the curation team at [cellxgene@chanzuckerberg.com](mailto:cellxgene@chanzuckerberg.com) with an email containing the following information.
    - Title
    - Description
    - Contact: name and email
    - Publication/preprint DOI: The publication digital object identifier (DOI) for the protocol. If no pre-print nor publication exists, please write 'not applicable'.
    - URLs: any additional URLs for related data or resources, such as GEO or protocols.io - can be added later
    - Consortia (i.e. HCA)

2. The team confirms acceptance of your data.
3. You prepare your AnnData file and send the file to [cellxgene@chanzuckerberg.com](mailto:cellxgene@chanzuckerberg.com).
4. The team will upload your dataset to a private collection where you can review.
5. The team will make your dataset either: public or private. Private datasets need to be shared with the integration team for inclusion in the integrated object.

