---
date: "2018-05-03"
description: "If you have questions or issues to report please email data-help@humancellatlas.org."
path: "/help/help/help-and-faq"
title: "Get Help"
---

# Get Help

If you have questions or issues to report please
email [data-help@humancellatlas.org](mailto:data-help@humancellatlas.org).

## Join the Data Portal slack community

The Data Portal is a community-driven effort.

We use the Human Cell Atlas Slack workspace to communicate events, highlight new data, and help participants find
answers to their Data Portal questions.

To join the conversation:

1. Request an invitation to the HCA Slack from <hca@humancellatlas.com>.
2. Add the  **#dcp-users** channel and say hello.

# FAQs

## Data availability and use guidelines

### What is the Human Cell Atlas?

The HCA is a collaborative effort by an international group of scientists to create a comprehensive reference map of all
the cells in the human body. The HCA *Data Portal* and Data Coordination Platform was built in partnership with the Chan
Zuckerberg Initiative and continues to be developed by scientists and technologists at EMBL-EBI, the Broad Institute,
and UC Santa Cruz. [Learn more about the HCA project here](https://www.humancellatlas.org/).

### What kind of data is in the HCA?

The HCA data will eventually accept all types of single-cell data, but it currently contains single-cell and
single-nuclei RNA-sequencing data from Smart-seq2 and 10x v2 and v3 assays. While we are focused on curating a
collection of data from healthy human specimens, as we grow we are accepting data from a number of sources, including
model organisms and organoids, and both healthy and diseased samples. Stay tuned, the HCA is constantly evolving!

Currently, both Smart-seq2 and 10x processed data can be downloaded as expression or count matrices, respectively, using
the Data Browser or by navigating to the Project page.

### How do I get more details about projects, like protocols or publications?

You can learn more about each project by visiting the Projects tab in the *Explore* section of the *Data Portal*. We
encourage all contributing projects to provide a detailed description of their protocol and post their protocol details
at [protocols.io](https://www.protocols.io/). Publication information, where available, is also visible on the Projects
page.

### Can I use this data for a publication?

Yes, data from the HCA can be used for publication. Learn more about
our [data use agreement](/about/data-use-agreement).

### How do I cite HCA data?

The URL for the HCA DCP *Data Portal*, https://data.humancellatlas.org/, can be used as the citation for data obtained
from the HCA. Alternatively, you can cite: Regev A., et al. 2017. The Human Cell Atlas. Elife. Dec 5;6. pii: e27041.
doi: 10.7554/eLife.27041.

## Data access, analysis, and development help

### How do I access the data in the HCA?

The user guides in the *Intro* section of the data portal contain detailed information
about [how to access the data](/guides).

### How do I visualize or analyze the data in the HCA?

Single-cell sequencing analysis methods are constantly changing, and there are many tools for visualizing or analyzing
the data in the HCA. Popular tools for analyzing single-cell RNA-seq data include Seurat (R) and ScanPy (Python),though
many other excellent tools exist.

### How do I make my tool compatible with HCA data and add it to the HCA Data Portal?

We encourage scientists and developers to contribute their own analysis and visualization applications and data
processing pipelines to the HCA. You can learn more about how to do this in the *Intro* section of the *Data Portal*.

### How can I get involved with the HCA?

Visit the *Contact* page to learn how to collaborate with us and to reach us with your feedback.

### How can I reuse parts of the DCP?

One of our goals is to make the HCA Data Coordination Platform code open and reusable to the community. For now, contact
us at data-help@humancellatlas.org with questions about reusing our code. Stay tuned for more information about reusing
code in the *Intro* section of the *Data Portal*.
